scipy/optimize/tests/test_least_squares.py

```diff
@@ -744,3 +746,28 @@ def test_basic():
     # test that 'method' arg is really optional
     res = least_squares(fun_trivial, 2.0)
     assert_allclose(res.x, 0, atol=1e-10)
+
+
+def test_small_tolerances_for_lm():
+    def res(p):
```

Comment:
Sorry, I forgot to mentions that, but now it seems that we don't need to use a complicated problem for this test. You can use `fun_trivial` (defined in this file) and `x` with two elements for example.

Reply:
No problem. I fixed it. PTAL.